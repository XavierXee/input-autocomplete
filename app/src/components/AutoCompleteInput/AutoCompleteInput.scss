@import '../../_variables.scss';

.AutoCompleteInput {
    position: relative;
    margin: 0;
    padding: $input-padding;
    color: $color_tertiary;
    box-sizing: border-box;
    width: 75%;
    font-family: inherit;
    font-size: $input-font-size;
    height: $input-height;
    border: solid 1px $color_tertiary;
}

.AutoCompleteInput::placeholder {
    color: $color_accent;
}

.AutoCompleteInput:focus {
    outline: none;
    box-shadow: 0.2rem 0.8rem 1.6rem white;
}