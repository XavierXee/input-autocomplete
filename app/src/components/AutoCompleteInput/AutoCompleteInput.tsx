import React from 'react';

import { AutoCompleteInputProps } from "../../interfaces/AutoCompleteInputProps";

import './AutoCompleteInput.scss';

class AutoCompleteInput extends React.Component<AutoCompleteInputProps> {
    constructor(props: any) {
        super(props);

        console.log(this.props);

        this.handleChange = this.handleChange.bind(this);
    }

    // TODO: type event
    handleChange(event: any): void {
        this.props.onChange(event.target.value);
    }

    render() {
        return (
            <input className="AutoCompleteInput" type="text" value={this.props.searchString} onChange={this.handleChange} />
        );
    }
}

export default AutoCompleteInput;
