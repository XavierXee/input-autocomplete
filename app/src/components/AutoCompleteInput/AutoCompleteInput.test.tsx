import React from 'react';
import ReactDOM from 'react-dom';
import AutoCompleteInput from './AutoCompleteInput';
import { create } from "react-test-renderer";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AutoCompleteInput searchString={'test'} onBlur={() => {}} onChange={() => {}} />, div);
});

it('render correctly', () => {
    const component = create(<AutoCompleteInput searchString={'test'} onBlur={() => {}} onChange={() => {}} />).toJSON();;
    expect(component).toMatchSnapshot();
});