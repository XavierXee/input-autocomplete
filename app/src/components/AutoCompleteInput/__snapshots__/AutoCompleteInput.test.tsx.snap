// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<span
  className="AutoCompleteInput"
>
  <i
    className="AutoCompleteInput--icon fas fa-search"
  >
    Â 
  </i>
  <input
    className="AutoCompleteInput--field"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value="test"
  />
</span>
`;
